docs/dev/ios/debug-glean-on-ios.md

```diff
@@ -0,0 +1,49 @@
+# Debug an iOS application against a different builds of Glean
+
+At times it may be necessary to debug against a local build of Glean or another git fork or branch in order to test new features or specific versions of Glean.
+
+Since Glean is consumed through [Carthage](https://github.com/Carthage/Carthage), this can be as simple as modifying the Cartfile for the consuming application.
+
+## Building against the latest Glean
+
+For consuming the latest version of Glean, the Cartfile contents would include the following line:
+
+```
+github "mozilla/glean" "master"
+```
+
+This will fetch and compile Glean from the [Mozilla/Glean GitHub](https://github.com/mozilla/glean/) repo from the "master" branch.
+
+## Building against a specific release of Glean
+
+For consuming a specific version of Glean, you can specify a branch name / tag name / commit name, like this:
+
+```
+github "mozilla/glean" "v0.0.1"
+```
+
+Where `v0.0.1` is a tagged release name, but could also be a branch name or a specific commit hash name like `832b222`
+
+If the custom Glean you wish to build from is a different fork on GitHub, you could simply modify the Cartfile to point at your fork like this:
+
+```
+github "myGitHubHandle/glean" "myBranchName"
```

Comment:
Woha, this is so much easier than the Gradle madness we need to go through with Android.

Reply:
Worth noting here that one can also use a local git repository?

```
git "file:///full/path/to/repositor" "branchName"
```

downside of carthage is that it can only work with commits

Reply:
I should start reading the whole thing.